### ISSUE TITLE
aiohttp server causing asyncio.create_subprocess_exec to hang

### ISSUE THREAD

**Issue by matclayton** [aio-libs/aiohttp#5336]
🐞 **Describe the bug**

When running a aiohttp server, upon calling 

```
process = await asyncio.create_subprocess_exec(
        *args,
        stdout=asyncio.subprocess.PIPE,
        stderr=asyncio.subprocess.PIPE,
        # stdin=asyncio.subprocess.DEVNULL,
    )
```

We never see this complete, outside of aiohttp this seems to work fine, and if we use DEVNULL for stdout and stderr it also appears to run the subprocess and output the results to the screen. I think something might be up with either the asyncio loop or the handling of pipes, but we're somewhat stumped by this. Any advice or suggestions would be welcome!

💡 **To Reproduce**

💡 **Expected behavior**

📋 **Logs/tracebacks**

No traceback

📋 **Your version of the Python**

3.9.1

📋 **Your version of the aiohttp/yarl/multidict distributions**

```
aiohttp == 3.7.3
...
```
```console
(3.9.0/envs/live-3.9.0) ➜  src git:(mc-transcode-manager) ✗ python -m pip show multidict        
Name: multidict
Version: 4.7.6
Summary: multidict implementation
Home-page: https://github.com/aio-libs/multidict
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /Users/mclayton/.pyenv/versions/3.9.0/envs/live-3.9.0/lib/python3.9/site-packages
Requires: 
Required-by: yarl, aiohttp
...
```
```console
$ python -m pip show yarl
Name: yarl
Version: 1.4.2
Summary: Yet another URL library
Home-page: https://github.com/aio-libs/yarl/
Author: Andrew Svetlov
Author-email: andrew.svetlov@gmail.com
License: Apache 2
Location: /Users/mclayton/.pyenv/versions/3.9.0/envs/live-3.9.0/lib/python3.9/site-packages
Requires: multidict, idna
Required-by: aiohttp
...
```

📋 **Additional context**

**Comment by matclayton** [comment]
After debugging a have a better PoC will open a new ticket, this is slightly misleading

**Comment by sadaszewski** [comment]
@matclayton I seem to be running into a similar problem. Could you please share what you have found and how you have finally solved the issue? Thank you in advance.

**Comment by matclayton** [comment]
Sorry I can't remember any of the context for this...

**Comment by Dreamsorcerer** [comment]
Well, the only other issue opened was #5338, so check that one.

**Comment by MisterLin1995** [comment]
> 

@sadaszewski  Any luck?
